src/main/java/com/example/simpleannotation/model/BuilderGeneratedClass.java

```diff
@@ -0,0 +1,181 @@
+import com.example.simpleannotation.model.descriptors.ConstructorDescriptor;
+import com.example.simpleannotation.model.descriptors.MethodDescriptor;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public final class BuilderGeneratedClass extends GeneratedClass {
+
+    private static final String BUILDER_NAME = "%sBuilder";
+    private static final String INSTANCE = "instance";
+    private static final String PUBLIC = "public";
+    private static final String FINAL = "final";
+    private static final String PRIVATE = "private";
+
+    private final BuilderAnnotatedClass annotatedClass;
+
+    public BuilderGeneratedClass(BuilderAnnotatedClass annotatedClass) {
+        this.annotatedClass = annotatedClass;
+        this.setClassName(computeBuilderName());
+        this.setPackageName(annotatedClass.getClassToBuildPackageName());
+        if (annotatedClass.useSingletonBuilder()) {
+            addSingletonPattern();
+        } else {
+            addSimpleConstructor();
+        }
+        addCommonFields();
+        if (annotatedClass.useFluentBuilder()) {
+            addFluentSetters();
+        } else {
+            addSimpleSetters();
+        }
+        addBuildMethod();
+    }
+
+    private void addCommonFields () {
+        this.getAttributes().add(new AttributeDescriptor("container", "Map<String, Object>", PRIVATE));
+        this.getImports().add("import java.util.Map;");
+        this.getImports().add("import java.util.HashMap;");
+        this.getImports().add(String.format("import %s.%s;",
+                annotatedClass.getClassToBuildPackageName(),
+                annotatedClass.getClassToBuild()));
+        this.getConstructors().forEach(constructor ->
+            constructor.addCodeLine("this.container = new HashMap<>();"));
+    }
+
+    private void addSingletonPattern() {
+        this.getAttributes().add(new AttributeDescriptor(INSTANCE,
+                computeBuilderName(),
+                "private static"));
+        MethodDescriptor descriptor = new MethodDescriptor("getInstance", computeBuilderName());
+        descriptor.addModifier(PUBLIC);
+        descriptor.addModifier("static");
+        descriptor.addCodeLine("if (instance == null) {")
+                .addCodeLine(String.format("\tinstance = new %s();", computeBuilderName()))
+                .addCodeLine("}")
+                .addCodeLine("return instance;");
+        this.getMethods().add(descriptor);
+        ConstructorDescriptor privateConstructor = new ConstructorDescriptor(computeBuilderName());
+        privateConstructor.addModifier(PRIVATE);
+        this.getConstructors().add(privateConstructor);
+    }
+
+    private void addSimpleConstructor() {
+        ConstructorDescriptor publicConstructor = new ConstructorDescriptor(computeBuilderName());
+        publicConstructor.addModifier(PUBLIC);
+        this.getConstructors().add(publicConstructor);
+    }
+
+    private void addFluentSetters () {
```

Comment:
delete extra space

Reply:
Done. Revised in [pull request 8](https://github.com/OscarCamacho/simpleannotation/pull/8)